README.md

```diff
@@ -1,66 +1,86 @@
+- The full list of your direct and transitive opam dependencies, according to the specifications
+  in the `.opam` files at your project's root, with hard version constraints (`{= <version>`) in
+  the `depends` field.
+- A `pin-depends` field filled with reproducible pins, either to tarballs or to git repos with an
+  explicit commit hash for all the above dependencies, except for `ocaml`, `dune` and any virtual
+  opam package.
+- A few extra fields meant to be interprated by `opam monorepo`
 
-       You can access the functionality directly via the monorepo-lock and
-       monorepo-pull commands,
+This lockfile can then be consumed by the following command:
+```
+opam monorepo pull
+```
+which will fetch the sources of all the locked dependencies into a `duniverse/` folder at the root of
+your project, marking them as `vendored_dirs` (see
+[dune's documentation](https://dune.readthedocs.io/en/latest/dune-files.html#vendored-dirs-since-1-11))
+so that dune will only build the artifacts you need from that folder.
 
-       Also see https://github.com/avsm/platform for an example of a fully
-       bootstrapping use of this tool.
+From that point you should be able to run `dune build` and `dune runtest` as you normally would and
+build your entire project from scratch!
 
-COMMANDS
-       lock
-           analyse opam files to generate a project-wide lock file
+### opam monorepo lock
 
-       pull
-           fetch the dependencies sources as specified by the lockfile
+It's important to note that `opam monorepo lock` will only succeed if all of your non-virtual and
+non "base" dependencies (e.g. `ocaml` or `dune`) build with dune (i.e. directly depend on the `dune`
+or `jbuilder` packages).
+If that's not the case the solver will report which packages don't build with dune.
 
-COMMON OPTIONS
-       --help[=FMT] (default=auto)
-           Show this help in format FMT. The value FMT must be one of `auto',
-           `pager', `groff' or `plain'. With `auto', the format is `pager` or
-           `plain' whenever the TERM env var is `dumb' or undefined.
+We maintain a [separate opam repository](https://github.com/dune-universe/opam-overlays) with dune
+ports of commonly used opam packages. If you have non-dune dependencies it is advised that you add
+this repository before running `opam monorepo lock`. You can do so by running the following command:
+```
+opam repository add dune-universe git+https://github.com/dune-universe/opam-overlays.git
+```
+Note that if it is not setup, the plugin will warn you.
 
-       --version
-           Show version information.
+The `lock` command takes your global and switch's opam configurations into account, meaning any
+opam repository or pins you set up will be picked up by the solver when resolving the full set of
+your project's dependencies.
 
-SEE ALSO
-       dune(1), git(1), opam(1)
+The generated lockfile is meant to be compatible with `opam` in such a way that running `opam
```

Comment:
Does that work? I was under the impression that this is a design goal but that's not 100% the case yet.

Reply:
It does! There might be a few corner cases where it doesn't but it definitely should work in most cases.